src/Features/Core/Portable/Completion/CompletionService.ProviderManager.cs

```diff
@@ -53,15 +62,34 @@ public ProviderManager(CompletionService service)
                 return _lazyImportedProviders;
             }
 
-            public static ImmutableArray<CompletionProvider> GetProjectCompletionProviders(Project? project)
+            private ValueTask ProcessBatchAsync(ImmutableSegmentedList<Project> projects, CancellationToken cancellationToken)
```

Comment:
correct me if i'm wrong.  but we're only using the project to get the .AnalyzerReferences to load the extensions from.  Can we relax all these apis to just take that IreadOnlyList?  another benefit of that is that we can not hold onto large solution graphs here.  we can also ask the work queue to remove duplicates.  so we'll generally only have a finite number of these to look at.

Reply:
Done.